### PR TITLE
Add cursor change after revert or stage changes

### LINKED ISSUE #75098: Return me to the last line of any reverted change
Step 0. Version 1.35.0 with extensions disabled.
Step 1. Open a file with changes committed in git
Step 2. Use ALT+F3 to navigate to the next change in-line
Step 3. Click Revert Change 
Step 4. find the line number you were on and move your cursor there, then hit ALT+F3 again

I would like vscode to remember where I was whenever I revert a change, then return me to that place, so that I can see the change disappear and so that I can proceed using hotkey to the next change.

### PR DESCRIPTION
**Pull request by Wowol**
This PR fixes #75098

### CODE REVIEW

**Comment by Wowol** [comment]
@joaomoreno it's been 2 months already, any update on this PR?

File: extensions/git/src/commands.ts [hunk 1]
```diff
@@ -980,7 +980,10 @@ export class CommandCenter {
 			return;
 		}
 
-		await this._stageChanges(textEditor, [changes[index]]);
+		await this._stageChanges(textEditor, [changes[index]]).then(() => {
+			const firstStagedLine = changes[index].modifiedStartLineNumber - 1;
+			textEditor.selections = [new Selection(firstStagedLine, 0, firstStagedLine, 0)];
+		});
```
Comment (joaomoreno) [inline extensions/git/src/commands.ts:986]:
```suggestion
		await this._stageChanges(textEditor, [changes[index]]);
		
		const firstStagedLine = changes[index].modifiedStartLineNumber - 1;
		textEditor.selections = [new Selection(firstStagedLine, 0, firstStagedLine, 0)];
```

File: extensions/git/src/commands.ts [hunk 2]
```diff
@@ -1027,7 +1030,10 @@ export class CommandCenter {
 			return;
 		}
 
-		await this._revertChanges(textEditor, [...changes.slice(0, index), ...changes.slice(index + 1)]);
+		await this._revertChanges(textEditor, [...changes.slice(0, index), ...changes.slice(index + 1)]).then(() => {
+			const firstStagedLine = changes[index].modifiedStartLineNumber - 1;
+			textEditor.selections = [new Selection(firstStagedLine, 0, firstStagedLine, 0)];
+		});
```
Comment (joaomoreno) [inline extensions/git/src/commands.ts:1036]:
```suggestion
		await this._revertChanges(textEditor, [...changes.slice(0, index), ...changes.slice(index + 1)]);
		
		const firstStagedLine = changes[index].modifiedStartLineNumber - 1;
		textEditor.selections = [new Selection(firstStagedLine, 0, firstStagedLine, 0)];
```

File: extensions/git/src/commands.ts [hunk 3]
```diff
@@ -1049,7 +1055,10 @@ export class CommandCenter {
 			return;
 		}
 
-		await this._revertChanges(textEditor, selectedChanges);
+		const selectionsBeforeRevert = textEditor.selections;
+		await this._revertChanges(textEditor, selectedChanges).then(() => {
+			textEditor.selections = selectionsBeforeRevert;
+		});
```
Comment (joaomoreno) [inline extensions/git/src/commands.ts:1061]:
```suggestion
		await this._revertChanges(textEditor, selectedChanges);
		textEditor.selections = selectionsBeforeRevert;
```

**Review by joaomoreno — COMMENTED** [review]
No need for `then`, since `await` is our friend.

**Review by joaomoreno — APPROVED** [review]

**Comment by joaomoreno** [comment]
Thanks!

**Pull request merged**